1000 fix empty selection in restore

m7pr opened the pull request:
Closes #1000 

Hey, I fixed the issue with disability to load a snapshot with empty `selected` field.
This filter can now be created, saved and uploaded.

However on an upload of such filter and selection of such filter the app goes back to empty filters. Is that intended?

chlebowa commented:
`selected = NULL` should set all values, `selected = character(0)` should set none.

chlebowa commented:
`selected = NULL` should set all values, `selected = character(0)` should set none.

m7pr commented:
Ok, just changed to character(0) and tested it sets filters to empty states which filter out all observations

averissimo commented on R/teal_slices-store.R:
```diff
@@ -75,6 +75,13 @@ slices_restore <- function(file) {
               slice[[field]]
             }
+        } else {
+          slice[[field]] <-
+            if (field == "selected") {
+              character(0)
+            } else {
+              NULL
+            }
```
:heart: the simplicity of solution. 1 line suggestion, but feel free to ignore

```suggestion
          slice[[field]] <- switch(field, selected = character(0), NULL)
```

chlebowa replied:
Don't you mean `slice[[field]] <- switch(field, "selected" = character(0), NULL)` :thinking:

averissimo reviewed (approved):
Great fix! :rocket: 

Solved the initial bug and tested a bunch of scenarios without being able to break.

The 1 line was broken into more :smiling_face_with_tear:

chlebowa commented:
So what about `"choices"`? If an app is started with `choices = character(0)` (for whatever reason) and then a snapshot is created, on upload it will be converted to `choices = NULL`, which means something different.

Should we worry about such a case?

m7pr commented:
Hey @chlebowa probably uncommon situation but I fixed the code so that it handles `character(0)` and `NULL`s separately. Also added 3 tests to verify that. Thanks for the input!
